### PR TITLE
Feat/[TM 1185] clip polygons 

### PR DESCRIPTION
**Pull request by egrojMonroy**
https://gfw.atlassian.net/browse/TM-1185

### CODE REVIEW

File: app/Helpers/PolygonGeometryHelper.php [hunk 1]
```diff
@@ -58,4 +58,13 @@ public static function updateProjectCentroidFromPolygon($polygonGeometry)
             Log::error('Error updating project centroid: ' . $e->getMessage());
         }
     }
+
+    public static function getPolygonsWithNames(array $uuids)
```
Comment (pachonjcl) [inline app/Helpers/PolygonGeometryHelper.php:62]: I don't think this method is going to be reused anytime soon the way it is, my suggestion would be to call this as `getPolygonsProjection` and pass parameters `$fields` that will be the current array of fields that you want to project your site polygon

Comment (pachonjcl) [reply app/Helpers/PolygonGeometryHelper.php:62]: Also I noted that your method does not take the `is_active` scope or filter into account. Please confirm that this is the intended behavior

File: app/Http/Controllers/V2/Terrafund/TerrafundClipGeometryController.php [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $polygonUuids = GeometryHelper::getSitePolygonsUuids($uuid)->toArray();
+
+        return $this->processClippedPolygons($polygonUuids);
+    }
+
+    public function clipOverlappingPolygons(string $uuid)
+    {
+        $polygonOverlappingExtraInfo = CriteriaSite::where('polygon_id', $uuid)
+            ->where('criteria_id', PolygonService::OVERLAPPING_CRITERIA_ID)
```
Comment (pachonjcl) [inline app/Http/Controllers/V2/Terrafund/TerrafundClipGeometryController.php:27]: create a scope in the CriteriaSite Model to only pass a CriteriaID as a parameter

**Review by pachonjcl — CHANGES_REQUESTED** [review]

File: app/Helpers/PolygonGeometryHelper.php (hunk 1 above)
Comment (egrojMonroy) [reply app/Helpers/PolygonGeometryHelper.php]: Changed to be more usefull

Comment (egrojMonroy) [reply app/Helpers/PolygonGeometryHelper.php]: also added is_active

File: app/Http/Controllers/V2/Terrafund/TerrafundClipGeometryController.php (hunk 2 above)
Comment (egrojMonroy) [reply app/Http/Controllers/V2/Terrafund/TerrafundClipGeometryController.php]: using a already created scope now

**Review by pachonjcl — APPROVED** [review]

**Pull request merged**